runtime/include/chpl-comm-compiler-macros.h

```diff
@@ -61,6 +62,28 @@ void chpl_gen_comm_get(void *addr, c_nodeid_t node, void* raddr,
     chpl_comm_get(addr, node, raddr, size, commID, ln, fn);
   }
 }
+#ifdef HAS_GPU_LOCALE
+static inline
+void chpl_gen_comm_get_gpu(void *addr, c_nodeid_t src_node,
```

Comment:
I am not opposed to your suggestion. I have a slight preference for keeping `chpl_gen_comm_get_` as the "prefix". How do you like `chpl_gen_comm_get_from_subloc` and `chpl_gen_comm_put_to_subloc` ? I can also use `gpu` instead of `subloc`. Though technically this function is not required to involve gpu-based allocations. So "aware" is a more precise word for sure. However, I can hide behind the fact that `c_sublocid_any` or `c_sublocid_none` are also a subloc :)

Reply:
> How do you like chpl_gen_comm_get_from_subloc and chpl_gen_comm_put_to_subloc ?

I actually prefer that on the basis that it future- proofs things for the theoretical future where we have other kinds of sublocs.

Reply:
https://github.com/chapel-lang/chapel/pull/22349/commits/05c8e80e50ee5589e49fa590a1c07a14365f0554